### PR TITLE
Add Firebase Segmentation SDK and some skeleton code in Firebase Android SDK

### PR DESCRIPTION
**Pull request by diwu-arete**

### CODE REVIEW

**Comment by diwu-arete** [comment]
kindly ping

File: firebase-segmentation/firebase-segmentation.gradle [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+plugins {
+    id 'firebase-library'
+    id 'com.google.protobuf'
+}
+
```
Comment (vkryachko) [inline firebase-segmentation/firebase-segmentation.gradle]: Also add the following pls:

```
firebaseLibrary {
    testLab.enabled = true
}
```

File: firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  /**
+   * Returns the {@link FirebaseSegmentation} initialized with the default {@link FirebaseApp}.
+   *
+   * @return a {@link FirebaseSegmentation} instance
+   */
+  @NonNull
+  public static FirebaseSegmentation getInstance() {
+    FirebaseApp defaultFirebaseApp = FirebaseApp.getInstance();
```
Comment (dmandar) [inline firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java]: might want to add a todo to check if default app is available, else throw error to user to call configure

Comment (diwu-arete) [reply firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java]: in that case, FirebaseApp.getInstance() will throw by itself.

**Comment by diwu-arete** [comment]
Re mandar: merging this branch to floc-master, good point!

File: firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java (hunk 2 above)
Comment (dmandar) [reply firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java]: sg

**Pull request merged**